app/src/main/java/sw19/moring03/paint/Views/DrawingView.java

```diff
@@ -65,31 +65,32 @@ public boolean onTouchEvent(MotionEvent event) {
     protected void onDraw(Canvas canvas) {
         super.onDraw(canvas);
 
-        if(objectsToPaint != null) {
-            for(Tools tool : objectsToPaint) {
-                tool.draw(canvas, mPaint);
+        if (objectsToPaint != null) {
+            for (Tools tool : objectsToPaint) {
+                tool.draw(canvas, paint);
             }
         }
     }
 
     private void addPoint(float x, float y) {
         PointF point = new PointF(x, y);
 
-        if(objectsToPaint.size() != 0) {
-            Tools currentTool = objectsToPaint.get(objectsToPaint.size()-1);
+        if (objectsToPaint.size() != 0) {
+            Tools currentTool = objectsToPaint.get(objectsToPaint.size() - 1);
 
-            if((currentTool instanceof PointTool) && ((PointTool)currentTool).getPointCount() > 0)
-            {
+            if ((currentTool instanceof PointTool) && ((PointTool) currentTool).getPointCount() > 0) {
                 return;
             }
 
-            objectsToPaint.get(objectsToPaint.size()-1).addPoint(point);
+            objectsToPaint.get(objectsToPaint.size() - 1).addPoint(point);
         }
     }
 
-    private void selectTool() {
+    public void selectTool() {
         Tool chosenTool = ((MainActivity) getContext()).getChosenTool();
 
+        Bitmap curBitmap = (((MainActivity) getContext()).newPhoto);
```

Comment:
Move this line to line 114 since you only need this variable in the TAKE_PHOTO case